pulsar-broker-common/src/main/java/org/apache/pulsar/broker/authentication/AuthenticationService.java

```diff
@@ -105,6 +105,18 @@ public String authenticateHttpRequest(HttpServletRequest request) throws Authent
         }
     }
 
+    public AuthenticationProvider getAuthenticationProvider(String authMethodName) {
+        return providers.get(authMethodName);
+    }
+
+    public String getAnonymousUserRole() throws AuthenticationException {
+        if (StringUtils.isNotBlank(anonymousUserRole)) {
+            return anonymousUserRole;
+        }
+        // If at least a provider was configured, then the authentication needs to be provided
```

Comment:
I don't see any checking here to see if a provider was configured.

Reply:
will remove this comments. this method is called in serverCnx, under condition: auth enabled, but authmethod="none". In this condition, it is expected anonymousUserRole provided.